pkg/controller/sparkapplication/sparkapp_util.go

```diff
@@ -95,29 +95,58 @@ func isDriverRunning(app *v1beta2.SparkApplication) bool {
 	return app.Status.AppState.State == v1beta2.RunningState
 }
 
-func driverStateToApplicationState(podStatus apiv1.PodStatus) v1beta2.ApplicationStateType {
+func getDriverTerminatedState(podStatus apiv1.PodStatus) *apiv1.ContainerStateTerminated {
```

Comment:
Rename `getDriverContainerTerminatedState`.
Two reasons:
* emphasize that we're digging down to containers
* otherwise it's confusing because it seems to indicate that the function returns a driver state, despite the return type.